Mathlib/Mathport/Rename.lean

```diff
@@ -54,11 +54,16 @@ def RenameMap.insert (m : RenameMap) (e : NameEntry) : RenameMap :=
 /-- Look up a lean 4 name from the lean 3 name. Also return the `dubious` error message. -/
 def RenameMap.find? (m : RenameMap) : Name → Option (String × Name) := m.toLean4.find?
 
+-- TODO: upstream into core/std
+instance [Inhabited α] : Inhabited (Thunk α) where
+  default := .pure default
+
 /-- This extension stores the lookup data generated from `#align` commands. -/
-initialize renameExtension : SimplePersistentEnvExtension NameEntry RenameMap ←
+-- wrap state in `Thunk` as we rarely actually query it
+initialize renameExtension : SimplePersistentEnvExtension NameEntry (Thunk RenameMap) ←
   registerSimplePersistentEnvExtension {
-    addEntryFn := (·.insert)
-    addImportedFn := mkStateFromImportedEntries (·.insert) {}
+    addEntryFn := fun t e => t.map (·.insert e)
```

Comment:
How costly is this, compared to keeping an array for the (unprocessed) entries in the current file? Intuitively, I would expect `Thunk.map` to need to do more work, but it does have the advantage that it can be mutated in place, which doesn't work for a pure data structure.

Reply:
On second thought, we need a list-like data structure anyway, since this is environment state which needs to be persistent, so a stack of closures is as good as any.

Reply:
`Thunk.map` is like, two allocations? Nothing worth counting I would hope if you `#align`s aren't in the millions.